modules/substrate/src/lib.rs

```diff
@@ -61,25 +59,36 @@ pub(crate) type BridgedBlockHash<T> = HashOf<<T as Trait>::BridgedChain>;
 /// Header of the bridged chain.
 pub(crate) type BridgedHeader<T> = HeaderOf<<T as Trait>::BridgedChain>;
 
+pub trait Trait: frame_system::Trait {
+	/// Chain that we are bridging here.
+	type BridgedChain: Chain;
+}
+
 decl_storage! {
 	trait Store for Module<T: Trait> as SubstrateBridge {
+		/// The number of the highest block(s) we know of.
+		ChainTipHeight: BridgedBlockNumber<T>;
 		/// Hash of the header at the highest known height.
-		BestHeader: BridgedBlockHash<T>;
+		///
+		/// If there are multiple headers at the same "best" height
+		/// this will contain all of their hashes.
+		BestHeaders: Vec<BridgedBlockHash<T>>;
```

Comment:
This has to do with a discussion from a few weeks ago on fork-choice rules. What I got out of it was that since we don't know what the fork-choice rule is for any given Substrate chain we should just default to picking the longest chain